server/historian/packages/historian-base/src/test/routes.spec.ts

```diff
@@ -44,384 +44,769 @@ const sendRequestsTillThrottledWithAssertion = async (superTest: request.SuperTe
+                });
+                it("/:ignored?/:tenantId/git/refs/*", async () => {
+                    await assertCorrelationId(`/repos/${tenantId}/git/refs/*`);
+                });
+                it("/:ignored?/:tenantId/git/refs post", async () => {
+                    await assertCorrelationId(`/repos/${tenantId}/git/refs`, "post");
+                });
+                it("/:ignored?/:tenantId/git/refs/* patch", async () => {
+                    await assertCorrelationId(`/repos/${tenantId}/git/refs/*`, "patch");
+                });
+                it("/:ignored?/:tenantId/git/refs/* delete", async () => {
+                    await assertCorrelationId(`/repos/${tenantId}/git/refs/*`, "delete");
+                });
             });
         });
-    });
 
-    describe("verify contents endpoints are throttled once throttling limit is exceeded", () => {
-        let app: express.Application;
-        let superTest: request.SuperTest<request.Test>;
-        let getContentStub: any;
-
-        beforeEach(() => {
-            getContentStub = sinon.stub(RestGitService.prototype, "getContent").returns(Promise.resolve({
-                sha: sha,
-                url: url,
-                tree: []
-            }));
-
-            const throttler = new TestThrottler(limit);
-            app = historianApp.create(
-                defaultProvider,
-                defaultTenantService,
-                defaultCache,
-                throttler
-            );
-            superTest = request(app);
-        });
+        describe("verify tags endpoints pass and store correlation id and add in response header", () => {
+            let getTagStub: any;
+            let createTagStub: any;
 
-        afterEach(() => {
-            getContentStub.restore();
-        });
+            beforeEach(() => {
+                getTagStub = sinon.stub(RestGitService.prototype, "getTag").returns(Promise.resolve({
+                    tag: "testTag",
+                    sha: sha,
+                    url: url,
+                    message: "testMessage",
+                    tagger: { name: "test", email: "test@domain.com", date: "now" },
+                    object: {
+                        type: "testType",
+                        sha: sha,
+                        url: url
+                    }
+                }));
+                createTagStub = sinon.stub(RestGitService.prototype, "createTag").returns(Promise.resolve({
+                    tag: "testTag",
+                    sha: sha,
+                    url: url,
+                    message: "testMessage",
+                    tagger: { name: "test", email: "test@domain.com", date: "now" },
+                    object: {
+                        type: "testType",
+                        sha: sha,
+                        url: url
+                    }
+                }));
+
+                const throttler = new TestThrottler(limit);
```

Comment:
For CorrelationID tests, you could probably pass a limit of `1000000` to ensure tests don't fail due to throttling

Reply:
Got it. Will add in the next update.